### ISSUE TITLE
Problem with Cookie domain path

### ISSUE THREAD

**Issue by sky4git** [jfarthing84/theme-my-login#46]
Why its required to set cookie domain path. Can't we just set cookie for password reset for domain level?

I am taking about cookie 
$rp_cookie = 'wp-resetpass-' . COOKIEHASH;

Cookie domain path is domainname/login. It gives problem using reset password with s2meember. As cookie path is not valid, it always redirect to 'invalid key' page.

**Comment by jfarthing84** [comment]
The cookie domain should be `/resetpass`. How are you getting `/login`?

**Comment by sky4git** [comment]
I am using s2member which I think overriding login URL. While we are
setting cookie for /resetpass path. Cookie is not being validated and thus
end up as invalid key error.

So if there is no particular reason, then I would suggest to set cookie at
domain level so problem of URL overriding can be avoided.

On Tuesday, 15 December 2015, Jeff Farthing notifications@github.com
wrote:

> The cookie domain should be /resetpass. How are you getting /login?
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/jfarthing84/theme-my-login/issues/46#issuecomment-164477161
> .

## 

Regards,
Aakash Dodiya

**Comment by jfarthing84** [comment]
I see no reason that removing the path would hurt. Have you tested if changing the path to '/' fixes the issue?

**Comment by sky4git** [comment]
Yes I have to fix like this

``` php
//list( $rp_path ) = explode( '?', wp_unslash( $_SERVER['REQUEST_URI'] ) );
$rp_path = '/';

Its working. Cookie domain path is just domain level here.
```

**Comment by sky4git** [comment]
More on the issue: I think since WordPress version 4.3.1, we cant send password to new user anymore. So WordPress send password set url to user, which goes to /login page.  Thus having a cookie path as /resetpass causes trouble.

**Comment by jfarthing84** [comment]
Fixed in eb2d09729da1c83156425e295ba6670192ef1019.
